digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/SpeedLimitService.scala

```diff
@@ -182,6 +182,12 @@ class SpeedLimitService(eventbus: DigiroadEventBus, vvhClient: VVHClient, roadLi
     }
   }
 
+  def hideUnknownSpeedLimits(linkIds: Set[Int]): Set[Int] = {
+    withDynSession {
```

Comment:
Since you are using an update, check if can be modified to this:
```suggestion
    withDynTransaction{
```